apm-agent-attach-cli/src/main/java/co/elastic/apm/attach/AgentDownloader.java

```diff
@@ -250,20 +284,31 @@ static TreeSet<Version> parseMavenMetadataXml(InputStream htmlInputStream) throw
     }
 
     /**
-     * Return the public key ID of our agent signing key.
+     * Returns the public keys used to sign agent artifacts
      *
-     * @return the public key ID
+     * @return map of signing keys, with key ID as key, and the raw public key value as value
      */
-    String getPublicKeyId() {
-        return "D27D666CD88E42B4";
+    public Map<String, byte[]> getPublicKeys() {
+        return keys;
     }
 
-    /**
-     * An input stream to the public key of the signing key.
-     *
-     * @return an input stream to the public key
-     */
-    InputStream getPublicKey() {
-        return AgentDownloader.class.getResourceAsStream("/public_key.asc");
+    private static byte[] getPubKeyContent(String path, int size) {
+        try (InputStream inputStream = AgentDownloader.class.getResourceAsStream(path)) {
+            if (inputStream == null) {
+                throw new IllegalStateException("unknown key file: " + path);
+            }
+            return toByteArray(inputStream, size);
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
     }
+
+    private static byte[] toByteArray(InputStream inputStream, int size) throws IOException {
+        byte[] result = new byte[size];
+        if (size == 0 || size != inputStream.read(result) || inputStream.read() >= 0) {
```

Comment:
[for reviewer] we can get away without a loop here as the content has known size and is rather small (less than 1kb).